web/src/util.ts

```diff
@@ -4,8 +4,26 @@ import * as blueslip from "./blueslip";
 import {$t} from "./i18n";
 import type {MatchedMessage, Message, RawMessage, UpdateMessageEvent} from "./types";
 
+// When the type is "private", properties from to_user_ids might be undefined.
+// See https://github.com/zulip/zulip/pull/23032#discussion_r1038480596.
+// Recipient type in direct messages (DM)
+export type DirectRecipient = {
+    type: "private";
+    to_user_ids: string;
+    reply_to: string;
+};
+
+// Recipient type in a stream
+export type StreamRecipient = {
+    type: "stream";
+    stream_id: number;
+    topic: string;
+};
+
 export type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<T>;
 
+type Recipient = PartialBy<DirectRecipient, "to_user_ids"> | StreamRecipient;
```

Comment:
Should we export this type and reuse it in `compose_fade_helper` ?

Reply:
`Recipient` in `util.ts` is slightly different from `compose_fade_helper`. `Recipient` in `util.ts` might expect `to_user_ids` to be undefined for reasons pointed in [this discussion](https://github.com/zulip/zulip/pull/23032#discussion_r1038480596).

`Recipient` in `compose_fade_users.js` expects a valid `to_user_ids` though.

Reply:
What makes it guaranteed that the `to_user_ids` will not be undefined for `focused_recipient` in `compose_fade_helper`?

This is where we set the `focused_recipient` for `compose_fade_helper`:-
https://github.com/zulip/zulip/blob/170a857bc13bdb902133950cd1476c3effe18fcb/web/src/compose_fade.js#L38-L42

But it looks like that `people.reply_to_to_user_ids_string` can return `undefined` as well so we might add an assertion here in case we are sure that here in this particular case this function will not return `undefined` value.

Reply:
Adding an assertion when setting its value makes sense. Fixed. Thanks for tracking it down